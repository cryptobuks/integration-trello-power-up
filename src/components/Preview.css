.SectionItem {
  border-radius: 3px;
  min-height: 5pc;
  margin: 0 0 0px;
  overflow: hidden;
  position: relative;
}

.SectionItem__Preview {
  border: 1px solid #d6dadc;
  background-color: #e2e4e6;
  background-position: center center;
  background-size: contain;
  background-repeat: no-repeat;
  border-radius: 3px;
  height: 78px;
  margin-top: -40px;
  position: absolute;
  top: 50%;
  left: 0;
  text-align: center;
  text-decoration: none;
  z-index: 0;
  width: 81pt;
}

.SectionItemDetails {
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  cursor: pointer;
  padding: 8px 0px 0px 90pt;
  min-height: 5pc;
  margin: 0;
}

.SectionItemDetails .SectionItemDetails__Name {
  font-weight: 700;
  line-height: 20px;
  padding: 2px 0;
  word-wrap: break-word;
}

.SectionItemDetails__Actions {
  margin-bottom: 8px;
}

.SectionItemDetails__Actions span:not(:first-child):before {
  content: ' - ';
}
