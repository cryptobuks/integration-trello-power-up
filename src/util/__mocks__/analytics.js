const analytics = {
  track: jest.fn(),
  props: {
    comment: 'comment',
    preview: 'preview',
    inspect: 'inspect',
    coverAdded: 'added',
    coverRemoved: 'removed'
  },
  names: {
    linkClicked: 'Trello.Link.Clicked',
    cardCover: 'Trello.CardCover.Changed',
    attachmentChanged: 'Trello.Attachment.Changed',
    attachmentViewed: 'Trello.Attachment.Viewed'
  },
  isUDFShareUrl: jest.fn()
}
export default analytics
