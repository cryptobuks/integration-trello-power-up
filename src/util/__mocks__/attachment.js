const attachment = jest.genMockFromModule('../attachment.js')
attachment.meta.mockReturnValue(
  Promise.resolve({
    projectName: 'Mobile App Prototype',
    companyName: '',
    companyDomain: 'https://projects.invisionapp.com',
    screenName: 'Compose - Blank',
    screenCount: 28,
    previewUrl: 'https://projects.invisionapp.com/preview',
    commentUrl: 'https://projects.invisionapp.com/comments',
    inspectUrl: 'https://projects.invisionapp.com/inspect',
    userName: 'Yauhen Ivashkevich',
    lastUpdatedAt: 1503077755000,
    mobileDeviceName: 'iPhone',
    shareUrl: 'https://invis.io/28D3U4ZCP',
    previewImgUrl: 'https://s3.invisionapp-cdn.com/some.png'
  })
)
attachment.remove.mockReturnValue(Promise.resolve({}))

export default attachment
