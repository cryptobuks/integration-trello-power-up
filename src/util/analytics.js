import config from './config'
// When the app is initializing, window.analytics doesn't exist yet, so we can't check whether it
// exists now. Instead, we check it on-demand.
const serverType = config.serverType
const Analytics = {
  track: (name, props) => {
    if (
      window.measure &&
      window.measure.collect &&
      typeof window.measure.collect === 'function'
    ) {
      window.measure.collect(name, props)
    }
  },
  isUDFShareUrl (shareUrl) {
    return shareUrl.match(/^https:\/\/(([a-z0-9-]+(\.[a-z0-9-]+)*?\.)?(invisionapp\.com|invision\.works)\/prototype)\/[A-Za-z-0-9]*(\/(preview|play|comment|inspect))?/i)
  },
  props: {
    thumbnail: { linkClicked: 'thumbnail', serverType },
    inspect: { linkClicked: 'inspect', serverType },
    preview: { linkClicked: 'preview', serverType },
    comment: { linkClicked: 'comment', serverType },
    removeUDF: { attachmentAction: 'remove', documentType: 'udf', serverType },
    removePrototype: {
      attachmentAction: 'remove',
      documentType: 'prototype',
      serverType
    },
    attachUDF: { attachmentAction: 'attach', documentType: 'udf', serverType },
    attachPrototype: {
      attachmentAction: 'attach',
      documentType: 'prototype',
      serverType
    },
    invalidLink: {
      attachmentAction: 'invalidLink',
      serverType
    },
    coverAdded: {
      coverAction: 'added'
    },
    coverRemoved: {
      coverAction: 'removed'
    },
    viewEmpty: { documentType: 'none', serverType },
    viewPrototype: { documentType: 'prototype', serverType },
    viewUDF: { documentType: 'udf', serverType }
  },
  names: {
    linkClicked: 'Trello.Link.Clicked',
    cardCover: 'Trello.CardCover.Changed',
    attachmentChanged: 'Trello.Attachment.Changed',
    attachmentViewed: 'Trello.Attachment.Viewed'
  }
}

export default Analytics
