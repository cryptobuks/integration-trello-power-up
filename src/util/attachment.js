import config from './config.js'
import T from './trello.js'

export function meta (url) {
  return new T.Promise((resolve, reject) => {
    fetch(`${config.baseUrl}/meta-bff/share?shareurl=${encodeURIComponent(url)}`).then(res => {
      if (res.ok) {
        resolve(res.json())
      } else {
        reject(res.statusText)
      }
    })
  })
}

export function remove (token, card, attachment) {
  return new T.Promise((resolve, reject) => {
    fetch(
      `https://api.trello.com/1/cards/${card}/attachments/${attachment}/?key=${config.clientKey}&token=${token}`,
      {
        method: 'DELETE'
      }
    ).then(res => {
      if (res.ok) {
        resolve(res.json())
      } else {
        reject(res.statusText)
      }
    })
  })
}

const attachment = { meta, remove }
export default attachment
