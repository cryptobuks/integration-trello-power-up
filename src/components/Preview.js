import React from 'react'
import moment from 'moment'
import cover from '../util/cover.js'
import T from '../util/trello.js'
import analytics from '../util/analytics.js'
import SectionItemLink from './SectionItemLink.js'
import './Preview.css'

export default class Preview extends React.Component {
  constructor (props) {
    super(props)
    this.t = T.iframe()
    this.makeCover = this.makeCover.bind(this)
    this.removeCover = this.removeCover.bind(this)
    this.getActions = this.getActions.bind(this)
    this.getLinks = this.getLinks.bind(this)
  }

  makeCover (event) {
    event.preventDefault()
    analytics.track(analytics.names.cardCover, {
      ...analytics.cover.added,
      ...this.props.documentType
    })
    return cover.make(this.props.meta).then(this.props.updateCover)
  }

  removeCover (event) {
    event.preventDefault()
    analytics.track(analytics.names.cardCover, {
      ...analytics.cover.removed,
      ...this.props.documentType
    })
    return cover.remove(this.props.meta).then(this.props.updateCover)
  }

  getActions () {
    let actions = [
      {
        handler: this.props.disconnect,
        text: 'Disconnect prototype',
        className: 'disconnect'
      }
    ]
    if (!this.props.cover) {
      actions.unshift({
        handler: this.makeCover,
        text: 'Make cover',
        className: 'make-cover'
      })
    } else {
      actions.unshift({
        handler: this.removeCover,
        text: 'Remove from cover',
        className: 'remove-cover'
      })
    }
    return actions
  }
  getLinks () {
    return [
      {
        url: this.props.meta.commentUrl,
        text: 'Comment',
        className: 'comment',
        prop: {
          ...this.props.documentType,
          ...analytics.links.comment
        }
      },
      {
        url: this.props.meta.previewUrl,
        text: 'Preview',
        className: 'preview',
        prop: {
          ...this.props.documentType,
          ...analytics.links.preview
        }
      },
      {
        url: this.props.meta.inspectUrl,
        text: 'Inspect',
        className: 'inspect',
        prop: {
          ...this.props.documentType,
          ...analytics.links.inspect
        }
      }
    ]
  }
  render () {
    const backgroundStyle = {
      backgroundImage: "url('" + this.props.meta.previewImgUrl + "')",
      backgroundPositionY: '0px',
      backgroundSize: 'cover'
    }

    return (
      <div className='SectionItem basic-attachment-list-item'>
        <a
          className='SectionItem__Preview'
          href={this.props.meta.shareUrl}
          target={this.props.meta.shareUrl}
          title={this.props.meta.screenName}
          style={backgroundStyle}
          onClick={() =>
            analytics.track(analytics.names.linkClicked, {
              ...this.props.documentType,
              ...analytics.links.thumbnail
            })
          }
        />
        <p className='SectionItemDetails attachment-thumbnail-details'>
          <span className='SectionItemDetails__Name'>
            {this.props.meta.screenName || this.props.meta.shareUrl}
          </span>
          <span className='u-block quiet SectionItemDetails__Actions'>
            {this.getLinks().map(link => {
              return (
                <SectionItemLink
                  {...link}
                  handler={() =>
                    analytics.track(analytics.names.linkClicked, link.prop)
                  }
                  key={this.props.attachment.id + link.className}
                />
              )
            })}
          </span>
          <span className='u-block quiet SectionItemDetails__Actions'>
            {this.getActions().map(action => {
              return (
                <SectionItemLink
                  {...action}
                  key={this.props.attachment.id + action.className}
                />
              )
            })}
          </span>
          <span
            className='attachment-thumbnail-details-title'
            href={this.props.meta.commentUrl}
            target={this.props.meta.commentUrl}
            title={this.props.meta.screenName}
          >
            <span className='u-block quiet'>
              Modified:
              <span
                className='date'
                title={moment(this.props.meta.lastUpdatedAt).fromNow()}
              >
                {' '}
                {moment(this.props.meta.lastUpdatedAt).fromNow()}
              </span>{' '}
              by {this.props.meta.userName}
            </span>
          </span>
        </p>
      </div>
    )
  }
}
