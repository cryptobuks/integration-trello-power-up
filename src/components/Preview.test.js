import 'jsdom-global/register'
import React from 'react'
import { mount } from 'enzyme'
import Promise from 'bluebird'
import sinon from 'sinon'
import toJson from 'enzyme-to-json'
import analytics from '../util/analytics.js'
import attachment from '../util/attachment.js'
// import auth from '../util/auth.js'
// import cover from '../util/cover.js'

import Preview from './Preview.js'

jest.mock('../util/trello.js')
jest.mock('../util/config.js')
jest.mock('../util/auth.js')
jest.mock('../util/cover.js')
jest.mock('../util/attachment.js')
jest.mock('../util/analytics.js')

const setup = cover => {
  const ctx = {}
  ctx.props = {
    cover: cover ? { id: 2, name: 'https://invis.io/28D3U4ZCP' } : undefined,
    attachment: {
      id: 1,
      url: 'https://invis.io/28D3U4ZCP'
    },
    meta: attachment.__meta,
    card: {
      id: 3
    },
    updateCover: jest.fn()
  }

  ctx.wrapper = mount(<Preview {...ctx.props} />, {
    disableLifecycleMethods: false
  })

  ctx.component = ctx.wrapper.instance()
  expect(ctx.component).toBeDefined()
  expect(toJson(ctx.wrapper)).toMatchSnapshot()
  return ctx
}

describe('Preview', () => {
  beforeAll(() => {
    sinon.spy(Preview.prototype, 'setState')
  })

  beforeEach(() => {
    Preview.prototype.setState.reset()
  })

  describe('Preview with cover', () => {
    it('should be able to track navigation clicks', () => {
      const ctx = setup(true)
      return Promise.attempt(() => {
        ctx.wrapper.find('a.comment').simulate('click')
        ctx.wrapper.find('a.preview').simulate('click')
        ctx.wrapper.find('a.inspect').simulate('click')
      }).then(() => {
        expect(analytics.track).toHaveBeenCalledWith('Trello.Link.Clicked', {
          ...analytics.links.inspect
        })
        expect(analytics.track).toHaveBeenCalledWith('Trello.Link.Clicked', {
          ...analytics.links.preview
        })
        expect(analytics.track).toHaveBeenCalledWith('Trello.Link.Clicked', {
          ...analytics.links.comment
        })
      })
    })
    it('should be able to remove cover', () => {
      const ctx = setup(true)
      return Promise.attempt(() => {
        ctx.wrapper.find('a.remove-cover').simulate('click')
      }).then(() => {
        expect(analytics.track).lastCalledWith('Trello.CardCover.Changed', {
          ...analytics.cover.removed
        })
      })
    })
    it('should be able disconnect prototype and cover', () => {
      const ctx = setup(true)
      return Promise.attempt(() => {
        ctx.wrapper.find('a.disconnect').simulate('click')
      })
    })
  })
  describe('Preview without cover', () => {
    it('should be able to set cover', () => {
      const ctx = setup(false)
      return Promise.attempt(() => {
        ctx.wrapper.find('a.make-cover').simulate('click')
      }).then(() => {
        expect(analytics.track).lastCalledWith('Trello.CardCover.Changed', {
          ...analytics.cover.added
        })
      })
    })
    it('should be able disconnect prototype and cover', () => {
      const ctx = setup(false)
      return Promise.attempt(() => {
        ctx.wrapper.find('a.disconnect').simulate('click')
      })
    })
  })
})
