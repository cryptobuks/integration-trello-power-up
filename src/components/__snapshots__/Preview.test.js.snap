// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview Preview with cover should be able disconnect prototype and cover 1`] = `
<Preview
  attachment={
    Object {
      "id": 1,
      "url": "https://invis.io/28D3U4ZCP",
    }
  }
  card={
    Object {
      "id": 3,
    }
  }
  cover={
    Object {
      "id": 2,
      "name": "https://invis.io/28D3U4ZCP",
    }
  }
  meta={
    Object {
      "commentUrl": "https://projects.invisionapp.com/comments",
      "companyDomain": "https://projects.invisionapp.com",
      "companyName": "",
      "inspectUrl": "https://projects.invisionapp.com/inspect",
      "lastUpdatedAt": 1503077755000,
      "mobileDeviceName": "iPhone",
      "previewImgUrl": "https://s3.invisionapp-cdn.com/some.png",
      "previewUrl": "https://projects.invisionapp.com/preview",
      "projectName": "Mobile App Prototype",
      "screenCount": 28,
      "screenName": "Compose - Blank",
      "shareUrl": "https://invis.io/28D3U4ZCP",
      "userName": "Yauhen Ivashkevich",
    }
  }
  updateCover={[Function]}
>
  <div
    className="SectionItem basic-attachment-list-item"
  >
    <a
      className="SectionItem__Preview"
      href="https://invis.io/28D3U4ZCP"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url('https://s3.invisionapp-cdn.com/some.png')",
          "backgroundPositionY": "0px",
          "backgroundSize": "cover",
        }
      }
      target="https://invis.io/28D3U4ZCP"
      title="Compose - Blank"
    />
    <p
      className="SectionItemDetails attachment-thumbnail-details"
    >
      <span
        className="SectionItemDetails__Name"
      >
        Compose - Blank
      </span>
      <span
        className="u-block quiet SectionItemDetails__Actions"
      >
        <SectionItemLink
          className="comment"
          handler={[Function]}
          name="comment"
          text="Comment"
          url="https://projects.invisionapp.com/comments"
        >
          <span>
            <a
              className="comment"
              href="https://projects.invisionapp.com/comments"
              onClick={[Function]}
              target="https://projects.invisionapp.com/comments"
            >
              Comment
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="preview"
          handler={[Function]}
          name="preview"
          text="Preview"
          url="https://projects.invisionapp.com/preview"
        >
          <span>
            <a
              className="preview"
              href="https://projects.invisionapp.com/preview"
              onClick={[Function]}
              target="https://projects.invisionapp.com/preview"
            >
              Preview
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="inspect"
          handler={[Function]}
          name="inspect"
          text="Inspect"
          url="https://projects.invisionapp.com/inspect"
        >
          <span>
            <a
              className="inspect"
              href="https://projects.invisionapp.com/inspect"
              onClick={[Function]}
              target="https://projects.invisionapp.com/inspect"
            >
              Inspect
            </a>
          </span>
        </SectionItemLink>
      </span>
      <span
        className="u-block quiet SectionItemDetails__Actions"
      >
        <SectionItemLink
          className="remove-cover"
          handler={[Function]}
          text="Remove from cover"
        >
          <span>
            <a
              className="remove-cover"
              href=""
              onClick={[Function]}
            >
              Remove from cover
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="disconnect"
          text="Disconnect prototype"
        >
          <span>
            <a
              className="disconnect"
              href=""
            >
              Disconnect prototype
            </a>
          </span>
        </SectionItemLink>
      </span>
      <span
        className="attachment-thumbnail-details-title"
        href="https://projects.invisionapp.com/comments"
        target="https://projects.invisionapp.com/comments"
        title="Compose - Blank"
      >
        <span
          className="u-block quiet"
        >
          Modified:
          <span
            className="date"
            title="5 months ago"
          >
             
            5 months ago
          </span>
           
          by 
          Yauhen Ivashkevich
        </span>
      </span>
    </p>
  </div>
</Preview>
`;

exports[`Preview Preview with cover should be able to remove cover 1`] = `
<Preview
  attachment={
    Object {
      "id": 1,
      "url": "https://invis.io/28D3U4ZCP",
    }
  }
  card={
    Object {
      "id": 3,
    }
  }
  cover={
    Object {
      "id": 2,
      "name": "https://invis.io/28D3U4ZCP",
    }
  }
  meta={
    Object {
      "commentUrl": "https://projects.invisionapp.com/comments",
      "companyDomain": "https://projects.invisionapp.com",
      "companyName": "",
      "inspectUrl": "https://projects.invisionapp.com/inspect",
      "lastUpdatedAt": 1503077755000,
      "mobileDeviceName": "iPhone",
      "previewImgUrl": "https://s3.invisionapp-cdn.com/some.png",
      "previewUrl": "https://projects.invisionapp.com/preview",
      "projectName": "Mobile App Prototype",
      "screenCount": 28,
      "screenName": "Compose - Blank",
      "shareUrl": "https://invis.io/28D3U4ZCP",
      "userName": "Yauhen Ivashkevich",
    }
  }
  updateCover={[Function]}
>
  <div
    className="SectionItem basic-attachment-list-item"
  >
    <a
      className="SectionItem__Preview"
      href="https://invis.io/28D3U4ZCP"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url('https://s3.invisionapp-cdn.com/some.png')",
          "backgroundPositionY": "0px",
          "backgroundSize": "cover",
        }
      }
      target="https://invis.io/28D3U4ZCP"
      title="Compose - Blank"
    />
    <p
      className="SectionItemDetails attachment-thumbnail-details"
    >
      <span
        className="SectionItemDetails__Name"
      >
        Compose - Blank
      </span>
      <span
        className="u-block quiet SectionItemDetails__Actions"
      >
        <SectionItemLink
          className="comment"
          handler={[Function]}
          name="comment"
          text="Comment"
          url="https://projects.invisionapp.com/comments"
        >
          <span>
            <a
              className="comment"
              href="https://projects.invisionapp.com/comments"
              onClick={[Function]}
              target="https://projects.invisionapp.com/comments"
            >
              Comment
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="preview"
          handler={[Function]}
          name="preview"
          text="Preview"
          url="https://projects.invisionapp.com/preview"
        >
          <span>
            <a
              className="preview"
              href="https://projects.invisionapp.com/preview"
              onClick={[Function]}
              target="https://projects.invisionapp.com/preview"
            >
              Preview
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="inspect"
          handler={[Function]}
          name="inspect"
          text="Inspect"
          url="https://projects.invisionapp.com/inspect"
        >
          <span>
            <a
              className="inspect"
              href="https://projects.invisionapp.com/inspect"
              onClick={[Function]}
              target="https://projects.invisionapp.com/inspect"
            >
              Inspect
            </a>
          </span>
        </SectionItemLink>
      </span>
      <span
        className="u-block quiet SectionItemDetails__Actions"
      >
        <SectionItemLink
          className="remove-cover"
          handler={[Function]}
          text="Remove from cover"
        >
          <span>
            <a
              className="remove-cover"
              href=""
              onClick={[Function]}
            >
              Remove from cover
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="disconnect"
          text="Disconnect prototype"
        >
          <span>
            <a
              className="disconnect"
              href=""
            >
              Disconnect prototype
            </a>
          </span>
        </SectionItemLink>
      </span>
      <span
        className="attachment-thumbnail-details-title"
        href="https://projects.invisionapp.com/comments"
        target="https://projects.invisionapp.com/comments"
        title="Compose - Blank"
      >
        <span
          className="u-block quiet"
        >
          Modified:
          <span
            className="date"
            title="5 months ago"
          >
             
            5 months ago
          </span>
           
          by 
          Yauhen Ivashkevich
        </span>
      </span>
    </p>
  </div>
</Preview>
`;

exports[`Preview Preview with cover should be able to track navigation clicks 1`] = `
<Preview
  attachment={
    Object {
      "id": 1,
      "url": "https://invis.io/28D3U4ZCP",
    }
  }
  card={
    Object {
      "id": 3,
    }
  }
  cover={
    Object {
      "id": 2,
      "name": "https://invis.io/28D3U4ZCP",
    }
  }
  meta={
    Object {
      "commentUrl": "https://projects.invisionapp.com/comments",
      "companyDomain": "https://projects.invisionapp.com",
      "companyName": "",
      "inspectUrl": "https://projects.invisionapp.com/inspect",
      "lastUpdatedAt": 1503077755000,
      "mobileDeviceName": "iPhone",
      "previewImgUrl": "https://s3.invisionapp-cdn.com/some.png",
      "previewUrl": "https://projects.invisionapp.com/preview",
      "projectName": "Mobile App Prototype",
      "screenCount": 28,
      "screenName": "Compose - Blank",
      "shareUrl": "https://invis.io/28D3U4ZCP",
      "userName": "Yauhen Ivashkevich",
    }
  }
  updateCover={[Function]}
>
  <div
    className="SectionItem basic-attachment-list-item"
  >
    <a
      className="SectionItem__Preview"
      href="https://invis.io/28D3U4ZCP"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url('https://s3.invisionapp-cdn.com/some.png')",
          "backgroundPositionY": "0px",
          "backgroundSize": "cover",
        }
      }
      target="https://invis.io/28D3U4ZCP"
      title="Compose - Blank"
    />
    <p
      className="SectionItemDetails attachment-thumbnail-details"
    >
      <span
        className="SectionItemDetails__Name"
      >
        Compose - Blank
      </span>
      <span
        className="u-block quiet SectionItemDetails__Actions"
      >
        <SectionItemLink
          className="comment"
          handler={[Function]}
          name="comment"
          text="Comment"
          url="https://projects.invisionapp.com/comments"
        >
          <span>
            <a
              className="comment"
              href="https://projects.invisionapp.com/comments"
              onClick={[Function]}
              target="https://projects.invisionapp.com/comments"
            >
              Comment
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="preview"
          handler={[Function]}
          name="preview"
          text="Preview"
          url="https://projects.invisionapp.com/preview"
        >
          <span>
            <a
              className="preview"
              href="https://projects.invisionapp.com/preview"
              onClick={[Function]}
              target="https://projects.invisionapp.com/preview"
            >
              Preview
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="inspect"
          handler={[Function]}
          name="inspect"
          text="Inspect"
          url="https://projects.invisionapp.com/inspect"
        >
          <span>
            <a
              className="inspect"
              href="https://projects.invisionapp.com/inspect"
              onClick={[Function]}
              target="https://projects.invisionapp.com/inspect"
            >
              Inspect
            </a>
          </span>
        </SectionItemLink>
      </span>
      <span
        className="u-block quiet SectionItemDetails__Actions"
      >
        <SectionItemLink
          className="remove-cover"
          handler={[Function]}
          text="Remove from cover"
        >
          <span>
            <a
              className="remove-cover"
              href=""
              onClick={[Function]}
            >
              Remove from cover
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="disconnect"
          text="Disconnect prototype"
        >
          <span>
            <a
              className="disconnect"
              href=""
            >
              Disconnect prototype
            </a>
          </span>
        </SectionItemLink>
      </span>
      <span
        className="attachment-thumbnail-details-title"
        href="https://projects.invisionapp.com/comments"
        target="https://projects.invisionapp.com/comments"
        title="Compose - Blank"
      >
        <span
          className="u-block quiet"
        >
          Modified:
          <span
            className="date"
            title="5 months ago"
          >
             
            5 months ago
          </span>
           
          by 
          Yauhen Ivashkevich
        </span>
      </span>
    </p>
  </div>
</Preview>
`;

exports[`Preview Preview without cover should be able disconnect prototype and cover 1`] = `
<Preview
  attachment={
    Object {
      "id": 1,
      "url": "https://invis.io/28D3U4ZCP",
    }
  }
  card={
    Object {
      "id": 3,
    }
  }
  meta={
    Object {
      "commentUrl": "https://projects.invisionapp.com/comments",
      "companyDomain": "https://projects.invisionapp.com",
      "companyName": "",
      "inspectUrl": "https://projects.invisionapp.com/inspect",
      "lastUpdatedAt": 1503077755000,
      "mobileDeviceName": "iPhone",
      "previewImgUrl": "https://s3.invisionapp-cdn.com/some.png",
      "previewUrl": "https://projects.invisionapp.com/preview",
      "projectName": "Mobile App Prototype",
      "screenCount": 28,
      "screenName": "Compose - Blank",
      "shareUrl": "https://invis.io/28D3U4ZCP",
      "userName": "Yauhen Ivashkevich",
    }
  }
  updateCover={[Function]}
>
  <div
    className="SectionItem basic-attachment-list-item"
  >
    <a
      className="SectionItem__Preview"
      href="https://invis.io/28D3U4ZCP"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url('https://s3.invisionapp-cdn.com/some.png')",
          "backgroundPositionY": "0px",
          "backgroundSize": "cover",
        }
      }
      target="https://invis.io/28D3U4ZCP"
      title="Compose - Blank"
    />
    <p
      className="SectionItemDetails attachment-thumbnail-details"
    >
      <span
        className="SectionItemDetails__Name"
      >
        Compose - Blank
      </span>
      <span
        className="u-block quiet SectionItemDetails__Actions"
      >
        <SectionItemLink
          className="comment"
          handler={[Function]}
          name="comment"
          text="Comment"
          url="https://projects.invisionapp.com/comments"
        >
          <span>
            <a
              className="comment"
              href="https://projects.invisionapp.com/comments"
              onClick={[Function]}
              target="https://projects.invisionapp.com/comments"
            >
              Comment
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="preview"
          handler={[Function]}
          name="preview"
          text="Preview"
          url="https://projects.invisionapp.com/preview"
        >
          <span>
            <a
              className="preview"
              href="https://projects.invisionapp.com/preview"
              onClick={[Function]}
              target="https://projects.invisionapp.com/preview"
            >
              Preview
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="inspect"
          handler={[Function]}
          name="inspect"
          text="Inspect"
          url="https://projects.invisionapp.com/inspect"
        >
          <span>
            <a
              className="inspect"
              href="https://projects.invisionapp.com/inspect"
              onClick={[Function]}
              target="https://projects.invisionapp.com/inspect"
            >
              Inspect
            </a>
          </span>
        </SectionItemLink>
      </span>
      <span
        className="u-block quiet SectionItemDetails__Actions"
      >
        <SectionItemLink
          className="make-cover"
          handler={[Function]}
          text="Make cover"
        >
          <span>
            <a
              className="make-cover"
              href=""
              onClick={[Function]}
            >
              Make cover
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="disconnect"
          text="Disconnect prototype"
        >
          <span>
            <a
              className="disconnect"
              href=""
            >
              Disconnect prototype
            </a>
          </span>
        </SectionItemLink>
      </span>
      <span
        className="attachment-thumbnail-details-title"
        href="https://projects.invisionapp.com/comments"
        target="https://projects.invisionapp.com/comments"
        title="Compose - Blank"
      >
        <span
          className="u-block quiet"
        >
          Modified:
          <span
            className="date"
            title="5 months ago"
          >
             
            5 months ago
          </span>
           
          by 
          Yauhen Ivashkevich
        </span>
      </span>
    </p>
  </div>
</Preview>
`;

exports[`Preview Preview without cover should be able to set cover 1`] = `
<Preview
  attachment={
    Object {
      "id": 1,
      "url": "https://invis.io/28D3U4ZCP",
    }
  }
  card={
    Object {
      "id": 3,
    }
  }
  meta={
    Object {
      "commentUrl": "https://projects.invisionapp.com/comments",
      "companyDomain": "https://projects.invisionapp.com",
      "companyName": "",
      "inspectUrl": "https://projects.invisionapp.com/inspect",
      "lastUpdatedAt": 1503077755000,
      "mobileDeviceName": "iPhone",
      "previewImgUrl": "https://s3.invisionapp-cdn.com/some.png",
      "previewUrl": "https://projects.invisionapp.com/preview",
      "projectName": "Mobile App Prototype",
      "screenCount": 28,
      "screenName": "Compose - Blank",
      "shareUrl": "https://invis.io/28D3U4ZCP",
      "userName": "Yauhen Ivashkevich",
    }
  }
  updateCover={[Function]}
>
  <div
    className="SectionItem basic-attachment-list-item"
  >
    <a
      className="SectionItem__Preview"
      href="https://invis.io/28D3U4ZCP"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url('https://s3.invisionapp-cdn.com/some.png')",
          "backgroundPositionY": "0px",
          "backgroundSize": "cover",
        }
      }
      target="https://invis.io/28D3U4ZCP"
      title="Compose - Blank"
    />
    <p
      className="SectionItemDetails attachment-thumbnail-details"
    >
      <span
        className="SectionItemDetails__Name"
      >
        Compose - Blank
      </span>
      <span
        className="u-block quiet SectionItemDetails__Actions"
      >
        <SectionItemLink
          className="comment"
          handler={[Function]}
          name="comment"
          text="Comment"
          url="https://projects.invisionapp.com/comments"
        >
          <span>
            <a
              className="comment"
              href="https://projects.invisionapp.com/comments"
              onClick={[Function]}
              target="https://projects.invisionapp.com/comments"
            >
              Comment
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="preview"
          handler={[Function]}
          name="preview"
          text="Preview"
          url="https://projects.invisionapp.com/preview"
        >
          <span>
            <a
              className="preview"
              href="https://projects.invisionapp.com/preview"
              onClick={[Function]}
              target="https://projects.invisionapp.com/preview"
            >
              Preview
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="inspect"
          handler={[Function]}
          name="inspect"
          text="Inspect"
          url="https://projects.invisionapp.com/inspect"
        >
          <span>
            <a
              className="inspect"
              href="https://projects.invisionapp.com/inspect"
              onClick={[Function]}
              target="https://projects.invisionapp.com/inspect"
            >
              Inspect
            </a>
          </span>
        </SectionItemLink>
      </span>
      <span
        className="u-block quiet SectionItemDetails__Actions"
      >
        <SectionItemLink
          className="make-cover"
          handler={[Function]}
          text="Make cover"
        >
          <span>
            <a
              className="make-cover"
              href=""
              onClick={[Function]}
            >
              Make cover
            </a>
          </span>
        </SectionItemLink>
        <SectionItemLink
          className="disconnect"
          text="Disconnect prototype"
        >
          <span>
            <a
              className="disconnect"
              href=""
            >
              Disconnect prototype
            </a>
          </span>
        </SectionItemLink>
      </span>
      <span
        className="attachment-thumbnail-details-title"
        href="https://projects.invisionapp.com/comments"
        target="https://projects.invisionapp.com/comments"
        title="Compose - Blank"
      >
        <span
          className="u-block quiet"
        >
          Modified:
          <span
            className="date"
            title="5 months ago"
          >
             
            5 months ago
          </span>
           
          by 
          Yauhen Ivashkevich
        </span>
      </span>
    </p>
  </div>
</Preview>
`;
